src/content/blog/definitelyformatted/index.mdx

```diff
@@ -0,0 +1,311 @@
+
+Still, as expected, some unexpected difficulties came up that we had to deal with.
+
+### Comment Directives
+
+Comments!
+For a language feature so many developers joke about never using, comments can be a surprisingly common annoying edge case.
+Especially when "comment directives" such as [`// eslint-ignore-next-line`](https://eslint.org/docs/latest/use/configure/rules#using-configuration-comments-1) (ESLint) and [`// @ts-expect-error`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments) (TypeScript) change the behavior of other tools.
+By far the most common cause of build breaks in the alphabet letter PRs was formatting changing which areas of code some comment directives applied to.
+
+Consider the following contrived example's `// @ts-expect-error`.
+The comment directive was originally correctly suppressing an error, but was auto-formatted to apply to the wrong line:
+
+```diff
+declare function onlyTakesNumbers(...input: number[]): void;
+
+// @ts-expect-error
+- onlyTakesNumbers(0, "this line should have // @ts-expect-error suppressing a type error", 1);
++ onlyTakesNumbers(
++  0,
++ "this line should have // @ts-expect-error suppressing a type error",
++  1
++ );
+```
+
+Formatters don't have any way to call into other tools such as ESLint or TypeScript to know what comments act as directives for which ranges of code.
+I had to manually move around quite a few comment directives in individual package PRs.
+The work wasn't practically difficult, but it was boring and tedious.
+
+### Merge Bot Changes
+
+The biggest change we made after the plan was on how to run the formatter for users.
+Our initial plan was to have the existing DefinitelyTyped merge bot apply dprint formatting to PRs before merging them.
+The bot was already creating commits on behalf of developers, so I'd hoped this would allow for keeping files formatted without bothering curmudgeonly developers.
+
+Unfortunately, the DefinitelyTyped merge bot wasn't set up to have state between _"PR is approved and passing builds"_ and _"PR is merging"_.
+We would need to add states like _"PR is being formatted"_, _"Formatting succeeded"_, and _"Formatting introduced build failures"_.
+
+We ended up opting for a simpler approach of solely relying on a `dprint fmt run` run on the default branch after each commit.
+It was much simpler to write and maintain than introducing new state flows to the existing merge bot.
+
+There is still the edge case that auto-formatting might introduce [comment directive](#comment-directives) CI failures to the main branch.
+We haven't seen that be a common pain point since this work went in.
+
+## The Results
```

Comment:
From @johnnyreilly:

> In "Results" it might be nice to summarise what was achieved:
> * Definitely Typed is now consistently formatted using the dprint tool
> * That consistent formatting is maintained by running dprint on each commit to master

Agreed!